torch/csrc/distributed/autograd/engine/dist_engine.cpp

```diff
@@ -175,6 +208,18 @@ void DistEngine::computeDependencies(
     }
   }
 
+  // Install graph task function pre hooks for all AccumulateGrad nodes to
+  // accumulate grads into the RPC context.
+  for (auto& pr : graphTask->exec_info_) {
+    if (auto accumulateGradFn = dynamic_cast<AccumulateGrad*>(pr.first)) {
+      auto& execInfo = pr.second;
+      execInfo.hooks_.push_back(std::make_unique<AccumulateGradPreHook>(
+          std::dynamic_pointer_cast<AccumulateGrad>(
+              accumulateGradFn->shared_from_this()),
+          autogradContext));
+    }
+  }
```

Comment:
We visit the AccumulateGrad nodes in the BFS above when we compute dependencies. Can we attach the post hooks at that point?